func/server.cc

```diff
@@ -14,22 +14,25 @@ Status FuncServiceImpl::unhook(ServerContext* context, const UnhookRequest* requ
   if (it == func_.end()) {
     return Status(StatusCode::NOT_FOUND, "Event not found");
   }
+  func_.erase(it);
   return Status::OK;
 }
 
 Status FuncServiceImpl::event(ServerContext* context, EventRequest* request, EventReply* reply) {
-/*  int event_type = request->event_type();
-  auto it = map_.find(event_type):
-  if (it == map_.end()) {
+  int event_type = request->event_type();
+  auto it = func_.find(event_type);
+  if (it == func_.end()) {
     return Status(StatusCode::NOT_FOUND, "Event not found");
   } 
-  std::string event_function = it*;
-  Any request_payload;
-  Any reply_payload;
-  request_payload.PackFrom(request->paylaod());
-  reply_payload(reply->payload());
-  warble_.Call(event_function, &request_payload, &reply_payload);
-*/
+  std::string event_function = it->second;
+  google::protobuf::Any request_payload;
+  google::protobuf::Any reply_payload;
+  request_payload.PackFrom(request->payload());
+  reply_payload.PackFrom(reply->payload());
+  
+  WarbleImpl warble;
+  warble.Call(event_function, &request_payload, &reply_payload);
```

Comment:
Correct me if I am wrong, but it appears that WarbleImpl currently has direct access to  the kvstore. This would indicate that the warble functions are not ephemeral.

One suggestion would be to pass the warble functions up to Func as pointers during a hook call and have Func make the calls to the  kvstore.

Reply:
oh yep right now it writes to a local map because i wanted to make sure it ran properly without the kvstore layer. I changed it now though so that it calls the client

Reply:
Okay, thanks for explaining. This does appear to be ephemeral as the kvstore client is being pulled up in each warble method. Moreover, the func_server only has the function map as its data member, which is good.

I still would recommend; however, that the warble functions not have direct access to the kvstore server. This may be too large of a change to make in the next couple of hours, but it could be an improvement to move the kvstore_client to be a member variable of func, and have calls to the kvstore be made there.

Reply:
This could even be accomplished by moving the Warble's RegisterUser, CreateWarble, etc. methods to a different file and have them solely exist in a namespace. The Warble Call() method could use these functions to package the calls correctly. Instead of the Warble functions making calls to the kvstore, they would simply return a value which Call() would use to get, put, and remove from the kvstore.